src/graph/executor/algo/SubgraphExecutor.cpp

```diff
@@ -100,17 +100,19 @@ folly::Future<Status> SubgraphExecutor::handleResponse(RpcResponse&& resps) {
 
 void SubgraphExecutor::filterEdges(int version) {
   auto iter = ectx_->getVersionedResult(subgraph_->outputVar(), version).iter();
-  auto* gnIter = static_cast<GetNeighborsIter*>(iter.get());
-  while (gnIter->valid()) {
-    const auto& dst = gnIter->getEdgeProp("*", nebula::kDst);
-    if (validVids_.find(dst) == validVids_.end()) {
-      auto edge = gnIter->getEdge();
-      gnIter->erase();
-    } else {
-      gnIter->next();
+  auto* gnIter = dynamic_cast<GetNeighborsIter*>(iter.get());
```

Comment:
Prefer to check the Kind of Iterator than using `dynamic_cast` to cast the type of object. dynamic_cast has runtime cost and maybe need to turn on the compile option `RTTI`, not recommend to use `dynamic_cast`.

Reply:
Used kind instead. Fixed.